autogen.sh

```diff
@@ -3,7 +3,7 @@
 autoreconf -v -f --install || exit 1
```

Comment:
Implicit function declarations were removed in C99, but it took until recently (unrelated to C23) for compilers to start enforcing it.

See https://wiki.gentoo.org/wiki/Modern_C_porting (which also features a list of other warnings you may want to enforce).